최창익/app/src/main/java/com/mash_up/mvvmstudy/view/detail/DetailAdapter.kt

```diff
@@ -0,0 +1,105 @@
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+
+        return when (viewType) {
+            0 -> ProfileViewHolder(ItemProfileBinding.inflate(inflater, parent, false))
+            1 -> DescriptionViewHolder(ItemDescriptionBinding.inflate(inflater, parent, false))
+            2 -> LanguageViewHolder(ItemLanguageBinding.inflate(inflater, parent, false))
+            else -> LastUpdatedViewHolder(ItemLastUpdatedBinding.inflate(inflater, parent, false))
+        }
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        val item = getItem(position)
+
+        when (holder) {
+            is ProfileViewHolder -> holder.bind(item as DetailFeed.Profile)
+            is LanguageViewHolder -> holder.bind(item as DetailFeed.Language)
+            is DescriptionViewHolder -> holder.bind(item as DetailFeed.Description)
+            is LastUpdatedViewHolder -> holder.bind(item as DetailFeed.LastUpdated)
+        }
+    }
+
+    class ProfileViewHolder(
+        private val binding: ItemProfileBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(profile: DetailFeed.Profile) {
+            binding.profile = profile
+        }
+    }
+
+    class LanguageViewHolder(
+        private val binding: ItemLanguageBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(profile: DetailFeed.Language) {
+            binding.language = profile
+        }
+    }
+
+    class DescriptionViewHolder(
+        private val binding: ItemDescriptionBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(profile: DetailFeed.Description) {
+            binding.description = profile
+        }
+    }
+
+    class LastUpdatedViewHolder(
+        private val binding: ItemLastUpdatedBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(profile: DetailFeed.LastUpdated) {
+            binding.lastUpdated = profile
+        }
+    }
+
+    companion object {
+        //TODO: Item에 id 값이 없는 경우 areContentTheSame과 areItemsTheSame이 같아지는 느낌..?
+        private val detailFeedDiffCallback = object : DiffUtil.ItemCallback<DetailFeed>() {
+            override fun areItemsTheSame(oldItem: DetailFeed, newItem: DetailFeed): Boolean {
+                when {
+                    oldItem is DetailFeed.Profile && newItem is DetailFeed.Profile && oldItem == newItem -> return true
+                    oldItem is DetailFeed.Description && newItem is DetailFeed.Description && oldItem == newItem -> return true
+                    oldItem is DetailFeed.Language && newItem is DetailFeed.Language && oldItem == newItem -> return true
+                    oldItem is DetailFeed.LastUpdated && newItem is DetailFeed.LastUpdated && oldItem == newItem -> return true
+                }
+
+                return false
+            }
```

Comment:
오.. 그렇군요
Item이 id가 없고 각 Data class의 필드가 하나뿐이라서 차이점이 사라지는군요... 흐음 저도 고민이네요

코드적으로는 아래처럼 줄일 수도 있을 것 같아요!
```suggestion
        //TODO: Item에 id 값이 없는 경우 areContentTheSame과 areItemsTheSame이 같아지는 느낌..?
        private val detailFeedDiffCallback = object : DiffUtil.ItemCallback<DetailFeed>() {
            override fun areItemsTheSame(oldItem: DetailFeed, newItem: DetailFeed): Boolean {
                    return if (oldItem.javaClass == newItem.javaClass) {
                    oldItem == newItem
                } else {
                    false
                }
```

Reply:
오 `.javaClass` 로 비교하니 훨씬 간결해지는군요 감사합니다!